mbbsd/mail.c

```diff
@@ -1261,24 +1322,29 @@ mail_account_sysop(void)
         return DIRCHANGED;
     }
 
-    for (i = 0; i < Vector_length(&namelist); i++) {
-        const char *userid = Vector_get(&namelist, i);
-        if(!searchuser(userid, idbuf))
-            continue;
-        sethomepath(genbuf, idbuf);
-        stampfile(genbuf, &mymail);
-        unlink(genbuf);
-        Copy(fpath, genbuf);
-
-        strlcpy(mymail.owner, cuser.userid, sizeof(mymail.owner));
-        strlcpy(mymail.title, save_title, sizeof(mymail.title));
-        mymail.filemode |= FILE_MULTI;
-        sethomedir(genbuf, idbuf);
-
-        if (append_record_forward(genbuf, &mymail, sizeof(mymail), idbuf) == -1)
-            vmsg(err_uid);
-        sendalert(idbuf, ALERT_NEW_MAIL);
+    const char *userid = Vector_get(&namelist, sysop_index);
+    if(!searchuser(userid, idbuf))
```

Comment:
A minor issue: it's bad user expierence that the user already wrote a lengthy mail and you told him the recipient is wrong and the mail is gone.